File: Assets/CSharpToUnityScript/aaTest/aaTest.js
import UnityEngine;
import System.Collections;
import System.Collections.Generic;

interface Int1 {}
interface Int2 {}
interface Int3 {}

// CLASSES DECLARATION
public class TestInheritance extends MonoBehaviour {}
public class TestInterface implements Int1 {
	public function TestInterface() {}
	public function TestInterface( aTruc: String) {}
	public function TestInterface( aTruc: boolean, aTruc2: boolean) {}
}
public class TestInterfaces implements Int1, Int2 {}
public class TestInterfaces2 implements Int1, Int2 , Int3 {}
public class aaTest extends TestInterface implements  Int1 ,Int2 {

	// CONSTRUCTORS
/*	public function aaTest( aTruc: String) {
super(aTruc);
		// struff there 1
	}

	public function aaTest() {
super();
		// struff there 2
	}

	public function aaTest( aTruc: boolean) {
aaTest(aTruc.ToString());
		// struff there 3
	}

	public function aaTest( mahcin: String) 
	{
aaTest();
		// struff there 4
	}
*/

	// ATTRIBUTES
/*
	
	
	@script RequireComponent(Material)
	@DrawGizmo(GizmoType.NotSelected | GizmoType.Pickable)
	@script AddComponentMenu("Transform/Follow Transform")
	@script ContextMenu("Do Something")
	@script MenuItem("MyMenu/Do Something")
	@script ExecuteInEditMode
	@script System.Serializable
*/


	// VARIABLES

	// multiple variable declaration
/*	var aVar9: double=var aVar11:  3.5, aVar10,=5.0 ;
	var aVar1: String = "un ;var aVar2:  espace",;
	var avar: TestInterface = new TestInterface()var avar2: , = new TestInterface (untruc);
*/	

	// VAR DECLARATION WITHOUT VALUE
	var withoutValue: String[];
	private var withoutValue1: boolean;
	private static var withoutValue2: String;
	static private var withoutValue3: Dictionary.<String, boolean>;


	// VAR DECLARATION WITH VALUE
	@HideInInspector
	public var withValue1: String = "mach ; in";
	public static var withValue2: String = "machin \\. "+	" machin chose"	;
	public static var withValue7: char = 'a'[0];
	static public var withValue3: boolean = false;
	protected var withValue4: float[] = [3.5, 0];
//	protected static var withValue5: double = Method (machin);
	static protected var withValue6: TestInterface = new TestInterface (withValue3, withValue3);

	
	// VAR DECLARATION IN FOREACH LOOP
	// se in function
	

	// CASTING
//	var varCast1: String = ReturnInt(withValue5);


	// ARRAYS

	var array1: String[];
	var array2: String[,] = new String[5,6];

	
	var array4: boolean[] = [ false, true];

	var array5: float[] =  [ 0.0,
		3.5,
		5];
	var array6: int[] = [ 0,
		3,
		5];

	// multi array
	var array7: int[,,] = new int[1,2,3];
	var array3: String[,] =  [ [ "","" ], [ "","" ] ]; // this is a jagged array
/*	
var array3Da: int[, ,] =  [ [ [ 1, 2, 3 ],  4: [, 5, 6 ] ], 
                                       [ [ 7, 8, 9 ],  10: [, 11, 12 ] ] ];

	// jagged arrays (it seems they don't quite work with UnityScript)
	var jagged: bool[][] = new bool[1][];
	var jaggedArray2: int[][] =  
	[
	     [1,3,5,7,9],
	     [0,2,4,6],
	     [11,22]
	];

	// jagged + multi
	var jaggedArray4: int[][,] =  
	[
	     [ [1,3], [5,7] ],
	     [ [0,2], [4,6], [8,10] ],
	     [ [11,22], [99,88], [0,9] ] 
	];
*/

	// FUNCTIONS
	@RPC
	public function ReturnVoid() {
		
		//function for(var _string: String in array1):
		for (var kvPair: KeyValuePair.<String, boolean> in withoutValue3) {}

		var testVar: String = "";
		if (true) {}
		else testVar = " ";

		//var _var: String = ReturnString (withValue3);
	}

	//private function ReturnString( testarg: boolean): String { return "";}

	static protected function ReturnBool( arg1: String): boolean {
		/*
		if (new Rect()) { // sould not be converted

		}
		else if (false) { // should not be converted

		}
*/
		arg1 = "";
		return false;
	}

	public override function ToString(): String {
		return "ToString()";
	}

	function ACoroutine( isTrue: boolean): IEnumerator {
		if (isTrue) {
			yield ;
		}
		else {
			yield  WaitForSeconds (3.5);
			// yield WaitForSeconds (3.5);
		}
		
	}



	// PROPERTIES
	private var prop1: int = 5;
	public  function get Prop1(): int {
	return prop1;
}
public function set Prop1(value: int) {
	prop1 = value;
}


	private var prop2: int;
	public  function get Prop2(): int { return prop2*2; 
		}


	private var prop3: String;
	public function set Prop3(value: String) {prop3   = value+" ";
		}


	private var prop4: double;
	public  function get Prop4(): double {return prop4; }
protected function set Prop4(value: double) {
			prop4 += value;
		}



	// Use this for initialization
	function Start() {
		var test: String ="";

		test = "machin";

		test = "";
		

		
	}


}

class AStruct extends System.ValueType {
	
}

File: Assets/CSharpToUnityScript/aaTest/aaTest.cs
using UnityEngine;
using System. Collections ;
using System. Collections . Generic;

interface Int1 {}
interface Int2 {}
interface Int3 {}
public class IParent {}

// CLASSES DECLARATION
public class TestInheritance : MonoBehaviour {}
public class TestIParent : IParent {}
public class TestInterface : Int1 {
	public TestInterface () {}
	public TestInterface (string aTruc) {}
	public TestInterface (bool aTruc, bool aTruc2) {}
}
public class TestInterfaces : Int1 ,Int2 {}
public class TestInterfaces2 : Int1 ,Int2 , Int3 {}
public class TestInterfaces3 : IParent ,Int2 , Int3 {}
public class aaTest : TestInterface, Int1 ,Int2 {

	// CONSTRUCTORS
/*	public aaTest (string aTruc) 
		: base (aTruc) {
		// struff there 1
	}

	public aaTest () : base () {
		// struff there 2
	}

	public aaTest (bool aTruc) 
		: this (aTruc.ToString()) {
		// struff there 3
	}

	public aaTest (string mahcin) 
		: this () 
	{
		// struff there 4
	}
*/

	// ATTRIBUTES
/*
	
	
	[RequireComponent ( typeof(Material) )]
	[DrawGizmo (GizmoType.NotSelected | GizmoType.Pickable)]
	[AddComponentMenu("Transform/Follow Transform")]
	[ContextMenu ("Do Something")]
	[MenuItem ("MyMenu/Do Something")]
	[ExecuteInEditMode]
	[System.Serializable]
*/


	// VARIABLES

	// multiple variable declaration
/*	double aVar9= 3.5f, aVar10, aVar11=5.0 ;
	string aVar1 = "un ; espace", aVar2;
	TestInterface avar = new TestInterface(), avar2 = new TestInterface (untruc);
*/	

	// VAR DECLARATION WITHOUT VALUE
	string[] withoutValue;
	private bool withoutValue1;
	private static string withoutValue2;
	static private Dictionary<string, bool> withoutValue3;


	// VAR DECLARATION WITH VALUE
	[HideInInspector]
	public string withValue1 = "mach ; in";
	public static string withValue2 = @"machin \. "+	" machin chose"	;
	public static char withValue7 = 'a';
	static public bool withValue3 = false;
	protected float[] withValue4 = {3.5f, 0f};
//	protected static double withValue5 = Method (machin);
	static protected TestInterface withValue6 = new TestInterface (withValue3, withValue3);

	
	// VAR DECLARATION IN FOREACH LOOP
	// se in function
	

	// CASTING
//	string varCast1 = (string)ReturnInt(out withValue5);


	// ARRAYS

	string[] array1;
	string[,] array2 = new string[5,6];

	
	bool[] array4 = { false, true};

	float[] array5 = new float[] { 0.0f,
		3.5f,
		5f};
	int[] array6 = { 0,
		3,
		5};

	// multi array
	int[,,] array7 = new int[1,2,3];
	string[,] array3 = new string[2,2] { { "","" }, { "","" } };
/*	
int[, ,] array3Da = new int[2, 2, 3] { { { 1, 2, 3 }, { 4, 5, 6 } }, 
                                       { { 7, 8, 9 }, { 10, 11, 12 } } };

	// jagged arrays (it seems they don't quite work with UnityScript)
	bool[][] jagged = new bool[1][];
	int[][] jaggedArray2 = new int[][] 
	{
	    new int[] {1,3,5,7,9},
	    new int[] {0,2,4,6},
	    new int[] {11,22}
	};

	// jagged + multi
	int[][,] jaggedArray4 = new int[3][,] 
	{
	    new int[,] { {1,3}, {5,7} },
	    new int[,] { {0,2}, {4,6}, {8,10} },
	    new int[,] { {11,22}, {99,88}, {0,9} } 
	};
*/

	// FUNCTIONS
	[RPC]
	public void ReturnVoid () {
		
		// VAR DELCARATION IN FOREACH LOOP
		foreach (string _string in array1) {}
		foreach (KeyValuePair<string, bool> kvPair in withoutValue3) {}

		string testVar = "";
		if (true) {}
		else
			testVar = " ";

		//string _var = ReturnString (withValue3);
	}

	//private string ReturnString (ref bool testarg) { return "";}

	static protected bool ReturnBool (out string arg1) {
		/*
		if (new Rect()) { // sould not be converted

		}
		else if (false) { // should not be converted

		}
*/
		arg1 = "";
		return false;
	}

	public override string ToString () {
		return "ToString()";
	}

	IEnumerator ACoroutine (bool isTrue) {
		if (isTrue) {
			yield return null;
		}
		else {
			yield return new WaitForSeconds (3.5f);
			// yield WaitForSeconds (3.5);
		}
		
	}



	// PROPERTIES
	private int prop1 = 5;
	public int Prop1 { get; set; }

	private int prop2;
	public int Prop2 {
		get { return prop2*2; 
		}
	}

	private string prop3;
	public string Prop3 {
		set 
		{ 
			prop3 = value+" ";
		}
	}

	private double prop4;
	public double Prop4 {
		get { return prop4; }
		protected set {
			prop4 += value;
		}
	}


	// Use this for initialization
	void Start () {
		string test ="";

		test = "machin";

		test = "";
		

		
	}


}

struct AStruct
{
	
}